### DISCUSSION TITLE
React Router <Link> Not Updating Page Content on Vercel Deployment despite having vercel.json file

### DISCUSSION THREAD

**Discussion by JoeDev2** [vercel/community#5026] [Help]
### Summary

Hello,

I'm encountering a perplexing issue with my React application deployed on Vercel. Despite using react-router-dom, the <Link> components are not functioning as expected. While the URL updates correctly when a <Link> is clicked, the page content does not refresh or update to display the new content. This issue is not present in my local development environment, where navigation works as expected. Also, just typing the URL does work since its causing a full refresh/navigation, and the <a> in the footer all work, so the router must work right? Because its correctly routing the / routes to the components, but the Links aren't working.

### Steps to Reproduce

{
  "rewrites": [{ "source": "/(.*)", "destination": "/" }]
}, 

Here's a brief overview of my setup:

Navbar Component (using LinkContainer

from react-router-bootstrap

):

import React from "react";

import { LinkContainer } from 'react-router-bootstrap';

import {Navbar, Nav, Image} from 'react-bootstrap';

import logo from '../assets/fanplay.svg';

import { Link } from 'react-router-dom';

import '../css/Navbar.css';

const LPNavBar = () => {

return (

<Navbar expand="lg" style={{backgroundColor: "#286491"}}>

<LinkContainer to="/home">

<Navbar.Brand className="navbar-brand-margin custom-logo">

<Image src={logo} width="30" height="30" className="d-inline-block align-top" />

</Navbar.Brand>

</LinkContainer>

<Navbar.Toggle aria-controls="basic-navbar-nav" />

<Navbar.Collapse id="basic-navbar-nav" className="justify-content-end">

<Nav>

<Nav.Link as={Link} to="/about">Home</Nav.Link>

<LinkContainer to="/">

<Nav.Link className="nav-link nav-link-text lp-navbar-extra-margin">Home</Nav.Link>

</LinkContainer>

<LinkContainer to="/about">

<Nav.Link className="nav-link nav-link-text lp-navbar-extra-margin">About</Nav.Link>

</LinkContainer>

AppRouter:

import React from 'react';

import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';

import HomePage from './pages/HomePage';

import UserProfilePage from './pages/UserProfilePage';

import LandingPage from './pages/LandingPage';

import AboutUs from "./pages/AboutUsPage";

const AppRouter = () => {

return (

<Router>

<Routes>

<Route path="/" element={<LandingPage />} />

<Route path="/home" element={<HomePage />} />

<Route path="/about" element={<AboutUs />} />

<Route path="/profile/:userId" element={<UserProfilePage />} />

</Routes>

</Router>

);

};

vercel.json in project root directory:

{

"rewrites": [

{ "source": "/(.*)", "destination": "/" }

]

}

Package.json (recently updated homepage to be "" but still doesn't work.)

{

"name": "app",

"version": "0.1.0",

"private": true,

"homepage": "",

"dependencies": {

"@fortawesome/fontawesome-svg-core": "^6.4.2",

"@fortawesome/free-solid-svg-icons": "^6.4.2",

"@fortawesome/react-fontawesome": "^0.2.0",

"@mui/icons-material": "^5.14.16",

"@stripe/react-stripe-js": "^2.1.1",

"@stripe/stripe-js": "^1.54.1",

"@testing-library/jest-dom": "^5.16.5",

"@testing-library/react": "^13.4.0",

"@testing-library/user-event": "^13.5.0",

"axios": "^1.4.0",

"bootstrap": "^5.3.0",

"react": "^18.2.0",

"react-bootstrap": "^2.7.4",

"react-bootstrap-icons": "^1.10.3",

"react-dom": "^18.2.0",

"react-icons": "^4.11.0",

"react-responsive-carousel": "^3.2.23",

"react-router-bootstrap": "^0.26.2",

"react-router-dom": "^6.12.1",

"react-scripts": "5.0.1",

"react-select": "^5.7.7",

"react-social-icons": "^6.6.0",

"styled-components": "^6.1.0",

"web-vitals": "^2.1.4"

},

"scripts": {

"start": "react-scripts start",

"build": "react-scripts build",

"test": "react-scripts test",

"eject": "react-scripts eject"

},

"eslintConfig": {

"extends": [

"react-app",

"react-app/jest"

]

},

"browserslist": {

"production": [

">0.2%",

"not dead",

"not op_mini all"

],

"development": [

"last 1 chrome version",

"last 1 firefox version",

"last 1 safari version"

]

},

"devDependencies": {

"raw-loader": "^4.0.2"

}

}

Troubleshooting Steps Taken:

Checked vercel.json

configuration for SPA routing.

Tried replacing LinkContainer

with the standard <Link>

component from react-router-dom

.

Ensured there are no custom event handlers interfering with the <Link>

behavior.

Checked for any network requests or console errors (none observed).

Examined the package.json

and found that the "homepage"

field was set to a specific URL, which was adjusted for testing.

Despite these efforts, the issue persists: <Link> updates the URL but doesn't trigger a page content update. This problem only occurs in the production build on Vercel; locally, everything works as expected.

Has anyone encountered a similar issue or have any suggestions on what else I could try? Any help or insight would be greatly appreciated!

**Comment by amyegan** [comment]
I wasn't able to replicate this `<Link>` problem in my example repo. Does this problem happen on localhost as well, or only when deployed? You may be able to get more information about the error from your browser console or local terminal.

If you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or link to a deployed project that has the problem, someone here might be able to spot the source of the problem.

My test repo: https://github.com/amyegan/react-router-5026
My deployed example: https://react-router-5026.vercel.app/
